Updating release notes for Keycloak 23 with some 'core features' impr…

mposolda opened the pull request:
…ovements

closes #23971

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

mposolda committed:
Updating release notes for Keycloak 23 with some 'core features' improvements
closes #23971

Co-authored-by: andymunro <48995441+andymunro@users.noreply.github.com>

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
 
 Both synced passkeys and device-bound passkeys can be used for both Same-Device and Cross-Device Authentication.
 However, passkeys operations success depends on the user's environment. Make sure which operations can succeed in https://passkeys.dev/device-support/[the environment].
+Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution and Thanks to https://github.com/thomasdarimont[Thomas Darimont] for the help with the
```
```suggestion
Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution and thanks to https://github.com/thomasdarimont[Thomas Darimont] for the help with the
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+
+= WebAuthn improvements
+
+There is new field `Extra Origins` added to WebAuthn policy, which allows better interoperability with non-Web platforms (e.g. native mobile applications).
```
```suggestion
WebAuthn policy now includes a new field: `Extra Origins`.  It provides better interoperability with non-Web platforms (for example, native mobile applications).
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+
+= More flexibility for introspection endpoint
+
+In previous versions, introspection endpoint returned automatically most of the claims, which were available in the access token. Now there is new
```
```suggestion
In previous versions, introspection endpoint  automatically returned most claims, which were available in the access token. Now there is new
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+= More flexibility for introspection endpoint
+
+In previous versions, introspection endpoint returned automatically most of the claims, which were available in the access token. Now there is new
+switch `Add to token introspection` on most of protocol mappers. This allows more flexibility as introspection endpoint can return different
```
```suggestion
switch `Add to token introspection` on most of protocol mappers. This addition allows more flexibility as introspection endpoint can return different
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+switch `Add to token introspection` on most of protocol mappers. This allows more flexibility as introspection endpoint can return different
+claims than access token. This is first step towards "Lightweight access tokens" support as access tokens can omit lots of the claims, which would be still returned
+by the introspection endpoint. When migrating from previous versions, the introspection endpoint should return same claims, which are returned from access token,
+so the behaviour should be effectively the same by default after the migration. Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution.
```
```suggestion
so the behavior should be effectively the same by default after the migration. Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution.
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+
+= Feature flag for OAuth 2.0 device authorization grant flow
+
+There is a feature flag added for OAuth 2.0 device authorization grant flow, so it is possible to easily disable this feature. Feature is still enabled by default.
```
```suggestion
OAuth 2.0 device authorization grant flow now includes a feature flag, so you can easily disable this feature. This eature is still enabled by default.
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+
+= Group scalability improvements
+
+Performance around searching of groups is improved for the use-cases with big amount of groups and subgroups. There are improvements, which allow
```
```suggestion
Performance around searching of groups is improved for the use-cases with many groups and subgroups. There are improvements, which allow
```

andymunro commented on docs/documentation/release_notes/topics/23_0_0.adoc:
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
+
+= User profile improvements
+
+Declarative user profile is still a preview feature in this release, but we are working hard on promoting it to supported feature. Feedback is welcome.
+If you find any issues or have any improvements in mind, it is welcome to create https://github.com/keycloak/keycloak/issues/new/choose[Github issue],
+ideally with label `area/user-profile`.
```
```suggestion
Declarative user profile is still a preview feature in this release, but we are working hard on promoting it to a supported feature. Feedback is welcome.
If you find any issues or have any improvements in mind,  you are welcome to create https://github.com/keycloak/keycloak/issues/new/choose[Github issue],
ideally with label `area/user-profile`.
```

andymunro reviewed (commented):
Thanks, @mposolda.  Just a few suggestions.

mposolda commented:
@andymunro Thanks! I've updated per your suggestions. I fixed minor typo in one of your suggestions (`This eature`, which should be `This feature`) and squashed the commits.
